web/modules/custom/usergreeting/config/install/usergreeting.settings.yml

```diff
@@ -0,0 +1,4 @@
+usergreeting_defaults:
```

Comment:
Besides providing a default settings file, it's also important to provide the schema for the stored data.

You can find examples in:

`web/core/modules/serialization/config/schema/serialization.schema.yml`
`web/core/modules/action/config/schema/action.schema.yml`

And there's a [comprehensive guide](https://www.drupal.org/docs/8/api/configuration-api/configuration-schemametadata) about them as well.

Apart from that, well done! :slightly_smiling_face: This PR looks very good.